### PR TITLE
Tesla zaps can now generate an energy ball if they zap a generator

### PR DESCRIPTION
**Pull request by lzimann**
:cl: Lzimann
add: Tesla zaps can now generate an energy ball if they zap a tesla generator!
/:cl:

This is probably unbalanced as fuck since I don't know how strong the power will be.

### CODE REVIEW

**Comment by kevinz000** [comment]
The next time I get antag I'm grabbing one and beelining the nearest electrified grille.

**Comment by coiax** [comment]
This will conflict with #23676

**Comment by PKPenguin321** [comment]
>The next time I get antag I'm grabbing one and beelining the nearest electrified grille.

To clarify Kevins vague comment, make sure tesla_act on this checks that it's a genuine bolt from a tesla, not from electrified grilles or whatever.

**Comment by lzimann** [comment]
Let's hope it does or this is going to be even funnier(and an even bigger bad idea)

File: code/modules/power/tesla/generator.dm [hunk 1]
```diff
@@ -3,4 +3,7 @@
 	desc = "Makes the wardenclyffe look like a child's plaything when shot with a particle accelerator."
 	icon = 'icons/obj/tesla_engine/tesla_generator.dmi'
 	icon_state = "TheSingGen"
-	creation_type = /obj/singularity/energy_ball
\ No newline at end of file
+	creation_type = /obj/singularity/energy_ball
+
+/obj/machinery/the_singularitygen/tesla/tesla_act(power, explosive = FALSE)
```
Comment (PKPenguin321) [inline code/modules/power/tesla/generator.dm]: you dont have to hope just set explosive to true

Comment (lzimann) [inline code/modules/power/tesla/generator.dm]: well this won't really do anything since the only thing this procs does is set the energy

**Comment by Iamgoofball** [comment]
it activating from grilles sounds like immersive gameplay

**Comment by anconfuzedrock** [comment]
only if you can spill the singulo drink on a generator to make one

**Pull request merged**